src/flb_utils.c

```diff
@@ -706,7 +706,10 @@ int flb_utils_write_str(char *buf, int *off, size_t size,
             encoded_to_buf(p, tmp, len);
             p += len;
         }
-        else if (c >= 0x80 && c <= 0xFFFF) {
+#if defined(__powerpc64__)
```

Comment:
This is the only part of this PR I want to understand better before approving it, I think I asked about it in the previous version but there was no response, could you explain why is this change necessary?

Reply:
Hi @leonardo-albertovich 
I remember you asked for this before. When I tried last time I could not get to the root of it in the limited time that was available. I made this change after the test failed, based on the actual value of c being received for Power. I will try to look at it again.

Reply:
Thank you, getting to the root of this is the only think missing from my point of view, if you need any help with it let me know, I'd be glad to assist you as much as possible.

Reply:
Hi @leonardo-albertovich
The issue is plain characters are signed on x86_64, and unsigned on ppc64le. Forcing chars on ppc64le to be signed using the gcc option **-fsigned-char**. So no change required to the original flb_utils.c. Will update the PR shortly.

Reply:
Awesome, with that out of the way I think it's good to go as long as it runs on CI and passes the tests of course.